gpt_index/docstore.py

```diff
@@ -127,6 +136,24 @@ def delete_document(
             raise ValueError(f"doc_id {doc_id} not found.")
         return doc
 
-    def __len__(self) -> int:
-        """Get length."""
-        return len(self.docs.keys())
+    def get_nodes(
+        self, node_ids: List[str], raise_error: bool = True
```

Comment:
out of curiosity why move these methods to docstore? originally i was thinking of making docstore as abstract as possible (so within docstore the methods would treat all values as the base document type)

Reply:
No particular reason other than it's easier to use.

I do think would be nice to keep DocumentStore generic, but I think there's larger problems like `contains_index_struct` right now ;)